gpytorch/priors/lkj_prior.py

```diff
@@ -1,93 +1,72 @@
+    r"""LKJ prior over n x n (positive definite) Cholesky-decomposed
+    correlation matrices
 
     .. math:
 
         \begin{equation*}
             pdf(\Sigma) ~ |\Sigma| ^ (\eta  - 1)
         \end{equation*}
 
-    where :math:`\eta > 0` is a shape parameter.
+    where :math:`\eta > 0` is a shape parameter and n is the dimension of the
+    correlation matrix.
 
-    Reference: Bayesian Data Analysis, 3rd ed., Gelman et al., p. 576
-    """
+    LKJCholeskyFactorPrior is different from LKJPrior in that it accepts the
+    Cholesky factor of the correlation matrix to compute probabilities.
 
-    arg_constraints = {"n": constraints.positive_integer, "eta": constraints.positive}
-    # TODO: move correlation matrix validation upstream into pytorch
-    support = constraints.positive_definite
-    _validate_args = True
+    This distribution assumes that the cholesky factor is lower-triangular.
+    """
 
-    def __init__(self, n, eta, validate_args=False):
+    def __init__(self, n, eta, validate_args=False, transform=None):
         TModule.__init__(self)
-        if not isinstance(n, int) or n < 1:
-            raise ValueError("n must be a positive integer")
-        if isinstance(eta, Number):
-            eta = torch.tensor(float(eta))
-        self.n = torch.tensor(n, dtype=torch.long, device=eta.device)
-        batch_shape = eta.shape
-        event_shape = torch.Size([n, n])
-        # Normalization constant(s)
-        i = torch.arange(n, dtype=eta.dtype, device=eta.device)
-        C = (((2 * eta.view(-1, 1) - 2 + i) * i).sum(1) * math.log(2)).view_as(eta)
-        C += n * torch.sum(2 * torch.lgamma(i / 2 + 1) - torch.lgamma(i + 2))
-        # need to assign values before registering as buffers to make argument validation work
-        self.eta = eta
-        self.C = C
-        super(LKJPrior, self).__init__(batch_shape, event_shape, validate_args=validate_args)
-        # now need to delete to be able to register buffer
-        del self.eta, self.C
-        self.register_buffer("eta", eta)
-        self.register_buffer("C", C)
+        LKJCholesky.__init__(self, dim=n, concentration=eta, validate_args=validate_args)
+        self.n = self.dim
+        self.eta = self.concentration
+        self._transform = transform
 
-    def log_prob(self, X):
-        if any(s != self.n for s in X.shape[-2:]):
-            raise ValueError("Correlation matrix is not of size n={}".format(self.n.item()))
-        if not _is_valid_correlation_matrix(X):
-            raise ValueError("Input is not a valid correlation matrix")
-        log_diag_sum = psd_safe_cholesky(X, upper=True).diagonal(dim1=-2, dim2=-1).log().sum(-1)
-        return self.C + (self.eta - 1) * 2 * log_diag_sum
+    def rsample(self, sample_shape=torch.Size()):
+        # mocking the sample call here
+        # TODO: determine why torch.distributions.LKJCholesky only implements sample
+        return super().sample(sample_shape=sample_shape)
```

Comment:
It seems like the proper thing to do here would be to overwrite the `sample` method and then raise a `NotImplementedError` for `rsample`. Or is there anywhere in the code where we're calling `rsample` where we really would be ok with calling `sample`?

Reply:
This was really a placeholder for checking into why https://github.com/pytorch/pytorch/pull/48798 uses sample instead of rsample (and also the name of the PR in the first place :) ).